.github/workflows/build.yml

```
@@ -37,3 +39,50 @@ jobs:
       - name: Run tests
         working-directory: ./v2
         run: go test ./...
+
+  create_release:
+    if: startsWith(github.ref, 'refs/tags/v')
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    needs: [lint, test]
+    steps:
+      - uses: actions/checkout@v4
+      - name: build and release
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release create ${{ github.ref_name }}
+  build_release:
```

```suggestion
  build_and_upload_to_github_release:
```

Abit verbose but I find this much clearer.